prow/tide/tide.go

```diff
@@ -461,38 +484,37 @@ func (c *Controller) query() (map[string]PullRequest, error) {
 		} else {
 			queries = map[string]string{"": query.Query()}
 		}
-
 		for org, q := range queries {
-			org, q, i := org, q, i
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				results, err := search(c.ghc.QueryWithGitHubAppsSupport, c.logger, q, time.Time{}, time.Now(), org)
-
-				resultString := "success"
-				if err != nil {
-					resultString = "error"
-				}
-				tideMetrics.queryResults.WithLabelValues(strconv.Itoa(i), org, resultString).Inc()
-
-				lock.Lock()
-				defer lock.Unlock()
-				if err != nil && len(results) == 0 {
-					c.logger.WithField("query", q).WithError(err).Warn("Failed to execute query.")
-					errs = append(errs, fmt.Errorf("query %d, err: %w", i, err))
-					return
-				}
-				if err != nil {
-					c.logger.WithError(err).WithField("query", q).Warning("found partial results")
-				}
-
-				for _, pr := range results {
-					prs[prKey(&pr)] = pr
-				}
-			}()
+			allQueries = append(allQueries, queryutil.QueryData{Query: q, Org: org, MetricIndex: i})
 		}
 	}
-	wg.Wait()
+	queryutil.QueriesInParallel(
+		c.config().Tide.MaxGraphQLGoroutines,
+		allQueries,
+		func(q queryutil.QueryData) {
+			results, err := search(c.ghc.QueryWithGitHubAppsSupport, c.logger, q.Query, time.Time{}, time.Now(), q.Org)
```

Comment:
For example - these lines (from `search` to the metrics update) must occur in *every* function that is passed to `QueriesInParallel` and are the bits that actually end up using the parameters in the query type.

Reply:
`search` from `blockers.go` is different than search from `tide.go`. I am afraid it can't be put to the same bucket as you want.

Reply:
So actually your finding here is not true. I found a time to look at this more closely. In `blockers.go` there is a need to query issues while in `tide.go` and `search.go` PRs (+ these are also dated queries). Maybe there can be found some kind of union of this, but may also result in Frankenstein function.
Metrics and error list is handled in both cases, but is this enough to create some kind of specific mechanism? Your comment^ may also apply to metrics and errors as well. Anyway, max that could be added to make above more specific is:
```
resultString := "success"
if err != nil {
    resultString = "error"
}
    tideMetrics.queryResults.WithLabelValues(strconv.Itoa(q.MetricIndex), q.Org, resultString).Inc()
```
Functions `search` are named the same, but the actual use cases of specific approaches are different (`tide.go`, `status.go` vs `blockers.go`, the body is different).
We had 2x queriesInPararell, one in `tide` package, second in `blockers`. We have now "too generic" one occurrence (sorry for that, this is how I interpreted your request). I get that you would like to unify more this code, but I don't see much more except the lines I pasted here.